elements/lisk-utils/src/job_handlers/mutex.ts

```diff
@@ -40,6 +40,29 @@ export class Mutex {
 		}
 	}
 
+	public async runExclusiveWithTimeout<T>(
+		worker: () => Promise<T>,
+		timeout: number,
+		timeoutMessage?: string,
+	): Promise<T> {
+		const release = await this.acquire();
+
+		const timeoutPromise = new Promise<T>((_, reject) => {
+			setTimeout(() => {
+				if (this.isLocked()) {
+					release();
```

Comment:
it is possible that this mutex is locked from other acquire, but release from the closure it locked